service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/clone/workspace/LaunchCloneGcsBucketResourceFlightStep.java

```diff
@@ -66,6 +75,12 @@ public StepResult doStep(FlightContext context) throws InterruptedException, Ret
         String.format("Clone GCS Bucket resource %s", resource.getResourceId().toString()));
     subflightInputParameters.put(
         ControlledResourceKeys.DESTINATION_RESOURCE_ID, destinationResourceId);
+    Map<String, String> destinationProperties = new HashMap<>(resource.getProperties());
+    if (destinationFolderId != null) {
+      destinationProperties.put(TERRA_FOLDER_ID_PROPERTY_KEY, String.valueOf(destinationFolderId));
+    }
+    subflightInputParameters.put(
+        ControlledResourceKeys.DESTINATION_RESOURCE_PROPERTY, destinationProperties);
```

Comment:
I see why you set the properties nullable now.

i think it's clearer if we just have a DESTINATION_FOLDER_ID instead of sometimes clearing properties sometimes not.

also pass in  @Nullable folderId instead of @Nullable properties into workspaceCloneUtils.